### DISCUSSION TITLE
RFC: improve `showMatchesPosition`

### DISCUSSION THREAD

**Discussion by flevi29** [meilisearch/product#787] [Feedback & Feature Proposal]
[`showMatchesPosition`](https://www.meilisearch.com/docs/reference/api/search#show-matches-position) is closely related to highlights, and it could be a better alternative. As of right now it doesn't respect [`searchableAttributes`](https://www.meilisearch.com/docs/learn/relevancy/displayed_searchable_attributes#searchable-fields), [`attributesToHighlight`](https://www.meilisearch.com/docs/reference/api/search#attributes-to-highlight), [`attributesToRetrieve`](https://www.meilisearch.com/docs/reference/api/search#attributes-to-retrieve), [`attributesToCrop`](https://www.meilisearch.com/docs/reference/api/search#attributes-to-crop), [`cropLength`](https://www.meilisearch.com/docs/reference/api/search#crop-length), but it could very well support all of these.
In fact `showMatchesPosition`, when enabled, should entirely replace `_formatted`, which should not be computed at all, this would further speed up Meilisearch, offloading the task of string manipulation to the frontend. If you're using `showMatchesPosition`, my thinking is that you're not touching `_formatted` either way.

Also, as far as I'm aware current implementation doesn't escape [highlight tags](https://www.meilisearch.com/docs/reference/api/search#highlight-tags), so there's always a slight chance that this could cause issues, but I haven't fully confirmed this just yet.
Also highlights are meant to be interpreted as HTML, so modern frameworks do a lot of work to sanitize it. This means that we could increase both backend and frontend performance with this.

I have previously worked on highlights (https://github.com/meilisearch/meilisearch/pull/4928), I would like to attempt to make this RFC a reality as well, if I know there is interest from the Meilisearch team.

**Comment by macraig** [comment]
Hi @flevi29 , thank you for your willingness to help us improve Meilisearch ❤️ We'll happily accept your contribution to improve the feature, let us know if you have any questions or need a review.
